src/css/main.less

```diff
@@ -635,3 +636,51 @@ fieldset[disabled] .btn-info.active {
     margin-left: 210px;
   }
 }
+
+.app-list-controls {
+    overflow: hidden;
+    padding-top: 8px;
+    padding-bottom: 8px;
+}
+
+.filter-box {
+    width: 330px;
```

Comment:
If we make this `max-width`, the filter input will shrink gracefully to fit the available screen width rather than forcing the button to wrap underneath.